main.go

```diff
@@ -26,17 +32,36 @@ func main() {
 	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
 	flag.Parse()
 
+	provider, err := New()
+	if err != nil {
+		return
+	}
+
+	var serveOpts []tf5server.ServeOpt
+
 	if debugMode {
-		plugin.Serve(&plugin.ServeOpts{
-			Debug:        true,
-			ProviderFunc: provider.New(),
-			ProviderAddr: "registry.terraform.io/hashicorp/hcp",
-		})
+		serveOpts = append(serveOpts, tf5server.WithManagedDebug())
+	}
 
-		return
+	err = tf5server.Serve("registry.terraform.io/hashicorp/hcp", provider, serveOpts...)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func New() (func() tfprotov5.ProviderServer, error) {
```

Comment:
I would add a comment here explaining the need for the muxing: 
- https://developer.hashicorp.com/terraform/plugin/framework/migrating
- https://developer.hashicorp.com/terraform/plugin/framework/migrating/mux

Reply:
Good idea!